push-build.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+if [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]
+then
+  BRANCH=$TRAVIS_PULL_REQUEST_BRANCH
+  BUILD="PR #${TRAVIS_PULL_REQUEST}"
+else
+  BRANCH=$TRAVIS_BRANCH
+  BUILD=$(git rev-parse --short HEAD)
+fi
+
+chmod 600 build_key
+eval `ssh-agent -s`
+ssh-add build_key
+
+cd _site
+git init
+git config user.name "Travis CI"
+git config user.email "builds@travis-ci.com"
+git remote add upstream "git@github.com:certbot/website-builds.git"
+
+# Branch from master to more easily compare across branches.
+git fetch upstream master
+git branch $BRANCH upstream/master
+# Manually set which branch HEAD points to to avoid changes to the
+# working tree caused by "git checkout".
+git symbolic-ref HEAD refs/heads/$BRANCH
+git reset
+
+# If this branch already exists upstream, use that instead.
+git fetch upstream $BRANCH
+git reset upstream/$BRANCH
```

Comment:
Perhaps my git fu isn't strong enough, but is there any difference in what's pushed to an existing branch using this approach rather than just using that branch above instead of `master`? Does the difference matter? If we changed the approach we'd have to check for the existence of the branch in the website-builds repo.

If this works cleanly as is though I'm fine with it.